lib/gpu/src/lib.rs

```diff
@@ -34,6 +35,13 @@ pub use shader::*;
 #[cfg(test)]
 mod basic_test;
 
+#[cfg(any(test, feature = "testing"))]
+lazy_static::lazy_static! {
+    pub static ref GPU_PANIC_MESSENGER: Box<dyn DebugMessenger> = Box::new(PanicIfErrorMessenger{});
+    pub static ref GPU_TEST_INSTANCE: std::sync::Arc<Instance> = Instance::builder().with_debug_messenger(GPU_PANIC_MESSENGER.as_ref()).build().unwrap();
+    pub static ref GPU_TEST_INSTANCE_NO_F16: std::sync::Arc<Instance> = Instance::builder().with_debug_messenger(GPU_PANIC_MESSENGER.as_ref()).build().unwrap();
```

Comment:
Both are the same?

Also, I suggest to remove `skip_half_precision` from the instance parameters and put it into the device constructor parameters.

Reply:
Removed parameter from the instance and applied it for the device only. In was type flag missing there, fixed